e2e/e2e_test.go

```diff
@@ -451,3 +451,138 @@ spec:
 		})
 	}
 }
+
+func TestPipeline(t *testing.T) {
+	var tests = []struct {
+		name     string
+		command  string
+		args     []string
+		input    map[string]string
+		expected map[string]string
+		errMsg   string
+	}{
```

Comment:
I am still thinking how we would like to structure these tests. We will have to do some work upfront to create a test harness that does the following:
 1. Takes input pkg (either some structured format specified in the `Go` test files or from a directory under `testData` on filesystem)
 2. A way to specify corresponding expected output (either some form in `Go` file or under a directory say `golden` under `testData`)
 3. Runs `pipeline run` how we are doing today where setting up the input dirs etc and then comparing the output to the golden files. I remember seeing utility method about comparing the two pkgs to be Equal or not in one of @mortent PR. 
 
Think, if we were to test pkgs with hierarchy that has sub-packages etc., how we can setup multiple such tests and invoke it with different pipeline setup.

Reply:
That makes sense, thanks. I will give more thought to this and would love to sync up if/when you have ideas for how these tests should be structured.

Reply:
I've updated the test structure to use directories testData and golden under testutils

Reply:
@mengqiy is setting up such e2e test harness with test packages that contain expected output/results.